clang-tools-extra/clangd/Compiler.cpp

```diff
@@ -89,6 +89,12 @@ void disableUnsupportedOptions(CompilerInvocation &CI) {
   CI.getLangOpts().ProfileListFiles.clear();
   CI.getLangOpts().XRayAlwaysInstrumentFiles.clear();
   CI.getLangOpts().XRayNeverInstrumentFiles.clear();
+  if (CI.getLangOpts().SYCLIsDevice) {
+    CI.getLangOpts().SYCLIsDevice = false;
+    CI.getLangOpts().SYCLUnnamedLambda = false;
+    CI.getLangOpts().DeclareSPIRVBuiltins = false;
+    CI.getTargetOpts().Triple = CI.getTargetOpts().HostTriple;
```

Comment:
Are there test cases you can add for these changes?

Reply:
Very good point @premanandrao, I agree this should be tested. Added a LIT test parsing basic SYCL code. The test reproduces the crash without these changes and succeeds with them.